Content.Server/Anomaly/Effects/PuddleCreateAnomalySystem.cs

```diff
@@ -0,0 +1,44 @@
+using Content.Server.Anomaly.Components;
+using Content.Server.Chemistry.EntitySystems;
+using Content.Shared.Anomaly.Components;
+using Content.Server.Fluids.EntitySystems;
+
+namespace Content.Server.Anomaly.Effects;
+
+/// <summary>
+/// This component allows the anomaly to create puddles from SolutionContainer.
+/// </summary>
+public sealed class PuddleCreateAnomalySystem : EntitySystem
+{
+
+    [Dependency] private readonly PuddleSystem _puddle = default!;
+    [Dependency] private readonly SolutionContainerSystem _solutionContainer = default!;
+
+    private EntityQuery<TransformComponent> _xformQuery;
+
+    public override void Initialize()
+    {
+        SubscribeLocalEvent<PuddleCreateAnomalyComponent, AnomalyPulseEvent>(OnPulse);
+        SubscribeLocalEvent<PuddleCreateAnomalyComponent, AnomalySupercriticalEvent>(OnSupercritical, before: new[] { typeof(SolutionContainerSystem), typeof(InjectionAnomalySystem)});
```

Comment:
Yes, the order of operation together with the injection component is important there: Otherwise, at first the anomaly splashes everything out in the form of puddles, and does not make any injection, since the reagent has run out.

Reply:
And reagent injections are a very important part that makes the anomaly dangerous.

Reply:
i understand the ordering for `InjectionAnomalySystem`, but not for `SolutionContainerSystem`

Reply:
removed. its working